src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceImplIntTest_updateReligion.kt

```diff
@@ -0,0 +1,274 @@
+    }
+
+    @Test
+    @Transactional(readOnly = true)
+    open fun `should not allow religion to be removed`() {
+      webTestClient.put()
+        .uri("api/offenders/A1234AA/religion")
+        .headers(setAuthorisation(listOf("ROLE_PRISON_API__PRISONER_PROFILE__RW")))
+        .header("Content-Type", APPLICATION_JSON_VALUE)
+        .bodyValue(NULL_RELIGION_UPDATE)
+        .exchange()
+        .expectStatus().isBadRequest
+        .expectBody()
+        .jsonPath("userMessage").isEqualTo("Malformed request")
+    }
+
+    @Test
+    @Transactional(readOnly = true)
+    open fun `should prevent invalid date format`() {
+      webTestClient.put()
+        .uri("api/offenders/A1234AA/religion")
+        .headers(setAuthorisation(listOf("ROLE_PRISON_API__PRISONER_PROFILE__RW")))
+        .header("Content-Type", APPLICATION_JSON_VALUE)
+        .bodyValue(INVALID_DATE_RELIGION_UPDATE)
+        .exchange()
+        .expectStatus().isBadRequest
+        .expectBody()
+        .jsonPath("userMessage").isEqualTo("Malformed request")
+    }
+
+    private fun religionProfileType(): ProfileType =
+      profileTypeRepository.findByTypeAndCategoryAndActive("RELF", "PI", true).get()
+  }
+
+  @Nested
+  inner class ErrorConditions {
+    @Test
+    fun `returns 404 when offender does not exist`() {
+      webTestClient.put()
+        .uri("api/offenders/AAA444/religion")
+        .headers(setAuthorisation(listOf("ROLE_PRISON_API__PRISONER_PROFILE__RW")))
+        .header("Content-Type", APPLICATION_JSON_VALUE)
+        .bodyValue(VALID_RELIGION_UPDATE)
+        .exchange()
+        .expectStatus().isNotFound
+        .expectBody().jsonPath("userMessage")
+        .isEqualTo("Prisoner with prisonerNumber AAA444 and existing booking not found")
+    }
+
+    @Test
+    fun `returns 404 when offender has no booking`() {
+      webTestClient.put()
+        .uri("api/offenders/A9880GH/religion")
+        .headers(setAuthorisation(listOf("ROLE_PRISON_API__PRISONER_PROFILE__RW")))
+        .header("Content-Type", APPLICATION_JSON_VALUE)
+        .bodyValue(VALID_RELIGION_UPDATE)
+        .exchange()
+        .expectStatus().isNotFound
+        .expectBody().jsonPath("userMessage")
+        .isEqualTo("Prisoner with prisonerNumber A9880GH and existing booking not found")
+    }
+
+    @Test
+    fun `returns 404 when a staff user account is not found for the provided username`() {
+      webTestClient.put()
+        .uri("api/offenders/A1234AA/religion")
+        .headers(setClientAuthorisation(listOf("ROLE_PRISON_API__PRISONER_PROFILE__RW")))
+        .header("Content-Type", APPLICATION_JSON_VALUE)
+        .bodyValue(VALID_RELIGION_UPDATE)
+        .exchange()
+        .expectStatus().isNotFound
+        .expectBody().jsonPath("userMessage")
+        .isEqualTo("Staff user account with provided username not found")
+    }
+
+    @Test
+    fun `returns status 423 (locked) when database row lock times out`() {
```

Comment:
I think this can be removed? We now don't expect that the locking condition will happen